### PR TITLE
[vote-program] adding instruction update validator identity

### PR DESCRIPTION
**Pull request by ochaloup**
I would like to use the `UpdateValidatorIdentity` instruction of VoteProgram in TS code and it is not available in the current web3.js library.
https://github.com/solana-labs/solana/blob/v1.18.4/sdk/program/src/vote/instruction.rs#L381

### CODE REVIEW

File: packages/library-legacy/src/programs/vote.ts [hunk 1]
```diff
@@ -535,9 +551,32 @@ export class VoteProgram {
   ): Transaction {
     if (params.lamports > currentVoteAccountBalance - rentExemptMinimum) {
       throw new Error(
-        'Withdraw will leave vote account with insuffcient funds.',
+        'Withdraw will leave vote account with insufficient funds.',
```
Comment (steveluscher) [inline packages/library-legacy/src/programs/vote.ts:554]: giphy

**Review by steveluscher — APPROVED** [review]
You're really going to like what we're working on next.

File: packages/library-legacy/src/programs/vote.ts (hunk 1 above)
Comment (steveluscher) [inline packages/library-legacy/src/programs/vote.ts:554]:
```suggestion
        'Withdraw will leave vote account with insuffcient funds.',
```

Comment (steveluscher) [reply packages/library-legacy/src/programs/vote.ts]: https://github.com/solana-labs/solana-web3.js/commit/e587f8d08dd981afe4fbcfac731e14e95d275027

**Comment by ochaloup** [comment]
> You're really going to like what we're working on next.

Thanks @steveluscher 

Could you tease me more details? Pretty interested :love_you_gesture: 
(btw. I was not able to get to finish my PR on vote accounts (https://github.com/solana-labs/solana-web3.js/pull/2013) and I trouble with stake accounts a bit as well. I would like potentially process through that while maybe not needed :-))

**Comment by steveluscher** [comment]
> Could you tease me more details? Pretty interested 🤟

Autogenerated program clients compatible with the new web3.js are moments away. cc/ @lorisleiva

**Pull request merged**